Example/S2MToolbox/Example/LocalNotificatinHelper/S2MNotificationViewController.m

```diff
@@ -0,0 +1,122 @@
+
+
+@implementation S2MNotificationViewController
+
+
+#pragma mark - Accessors
+
+- (void)setNotifications:(NSArray *)notifications
+{
+    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"s2mKey" ascending:YES];
+    _notifications = [notifications sortedArrayUsingDescriptors:@[sortDescriptor]];
+    [self.tableView reloadData];
+}
+
+#pragma mark - IBActions
+
+- (IBAction)removeAllNotifications:(id)sender
+{
+    [S2MNotificationHelper removeAllNotifications];
+    self.notifications = [S2MNotificationHelper allNotifications];
+}
+
+- (IBAction)reloadTableView:(id)sender
+{
+    self.notifications = [S2MNotificationHelper allNotifications];
+}
+
+- (IBAction)addLocalNotification:(id)sender
+{
+    NSString *keyForCache = [@([[NSDate date] timeIntervalSince1970]) stringValue];
+    UILocalNotification *notification = [UILocalNotification new];
+    notification.alertAction = @"Let's Check";
+    notification.alertBody = [NSString stringWithFormat:@"Test\n%@\n%@", [NSDate date], keyForCache];
+    notification.timeZone = [NSTimeZone defaultTimeZone];
+    [notification setS2mKey:keyForCache];
+    [S2MNotificationHelper showNotification:notification];
+    self.notifications = [S2MNotificationHelper allNotifications];
+}
+
+#pragma mark - UITableViewDataSource
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier];
+    UILocalNotification *noti = self.notifications[indexPath.row];
+    cell.textLabel.text = noti.alertBody;
+    cell.textLabel.numberOfLines = 0;
+    return cell;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return self.notifications.count;
+}
+
+#pragma mark - UITableViewDelegate
+
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [tableView deselectRowAtIndexPath:indexPath animated:NO];
+    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Remove Notification"
+                                                    message:@"Do you really want remove this notification?"
+                                                   delegate:self cancelButtonTitle:@"NO" otherButtonTitles:@"YES", nil];
+    alert.tag = indexPath.row;
+    [alert show];
+}
+
+#pragma mark - UIAlertViewDelegate
+
+- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex
+{
+    if (buttonIndex != alertView.cancelButtonIndex) {
+        [S2MNotificationHelper removeNotification:self.notifications[alertView.tag]];
+        self.notifications = [S2MNotificationHelper allNotifications];
+    }
+}
+
+#pragma mark - life cycle
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:CellIdentifier];
+    
+    UIApplication *application = [UIApplication sharedApplication];
```

Comment:
move request permission to viewDidAppear in order to get prompt when pushed animation is finished